app/models/user.rb

```diff
@@ -12,4 +13,26 @@ def User.digest string
     cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST : BCrypt::Engine.cost
     BCrypt::Password.create string, cost: cost
   end
-end
+
+  # Returns a random token.
+  def User.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  # Remembers a user in the database for use in persistent sessions.
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
```

Comment:
Một cái là update 1 attribute 1 cái là update nhiều attribute
Update attributes update dữ liệu thông qua băm và gói toàn bộ vào trong transaction, được bảo mật tốt hơn, nếu có lỗi sẽ rollback.
còn Update attribute chỉ lưu chữ record, không có validation, thường dùng để check cờ record